crates/serde/src/quantity.rs

```diff
@@ -164,6 +164,133 @@ pub mod u128_vec_vec_opt {
     }
 }
 
+/// Serde functions for encoding a hashmap of primitive numbers using the Ethereum "quantity"
+/// format.
+///
+/// See [`quantity`](self) for more information.
+pub mod hashmap {
+    use super::private::ConvertRuint;
+    use core::{fmt, marker::PhantomData};
+    use serde::{de::MapAccess, ser::SerializeMap, Deserializer, Serializer};
+    use std::collections::HashMap;
+
+    /// Serializes a hashmap of primitive numbers as a "quantity" hex string.
+    pub fn serialize<K, V, S>(map: &HashMap<K, V>, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
this should also relax the hasher generic, see for ref https://github.com/serde-rs/serde/blob/418062165f9fe395461db9f61569c3142c584854/serde/src/ser/impls.rs#L495-L506